test/interval.jl

```diff
@@ -125,12 +125,33 @@ isinf(::TimeType) = false
     end
 
     @testset "accessors" begin
-        for (a, b, _) in test_values
+        for (a, b, p) in test_values
             for (L, R) in BOUND_PERMUTATIONS
                 interval = Interval{L, R}(a, b)
 
                 @test first(interval) == a
                 @test last(interval) == b
+                # the value we compare to min/max depends on if the bound is open/closed
+                # for floats, adding/subtracting eps(Float) is inaccurate for large vals
+                # so we must use next/prev float
+
+                min_comp = if a in interval
+                    first(interval)
+                else
+                    typeof(first(interval)) <: AbstractFloat ?
+                        nextfloat(first(interval)) :
+                        first(interval) + p
```

Comment:
```suggestion
                    val = first(interval)
                    eltype(interval) isa AbstractFloat ? nextfloat(val) : val + p
```

Reply:
Actually, this extra logic can be removed if `AbstractFloat` respects explicitly set increment values